### ISSUE TITLE
有没有不用ray的版本

### ISSUE THREAD

**Issue by BBDrive** [llan-ml/tesp#4]
很欣赏您这篇文章，但是在复现的时候看这个代码真的好难懂，不知道有没有不用ray的版本，谢谢

**Comment by llan-ml** [comment]
@BBDrive 只有基于ray的实现哈~

**Comment by BBDrive** [comment]
您好，想问一下，RNN的input中的logits_buffer代表的是action的集合吗？

**Comment by llan-ml** [comment]
是policy预测的action distribution（这里是高斯分布的均值和方差的拼接）。

**Comment by BBDrive** [comment]
1.然后是在这个distribution下采取一个行动得到新的state，再把这个state放到mlp里吗？
2.这个过程结束的条件是该回合到达终点还是 收集到了一定数量的（s,a,r）
谢谢您的回答




------------------ 原始邮件 ------------------
发件人: lanlin <notifications@github.com&gt;
发送时间: 2019年10月10日 15:35
收件人: llan-ml/tesp <tesp@noreply.github.com&gt;
抄送: BBDrive <842089432@qq.com&gt;, Mention <mention@noreply.github.com&gt;
主题: 回复：[llan-ml/tesp] 有没有不用ray的版本 (#4)




是policy预测的action distribution（这里是高斯分布的均值和方差的拼接）。

**Comment by llan-ml** [comment]
1. RNN的输入的一个episode buffer中存的之前采样过的episode，是一个序列，每个元素是(state, logit, reward)的拼接。MLP的输入是当前的state和RNN输出的embedding，正常RL的采样还是用MLP进行的，只是MLP多了一个embedding作为输入。
2. 两个条件满足一个就行。
